*** orig/ruby/curia/mod_curia.c	2006-09-25 21:26:06.000000000 +0900
--- patched/ruby/curia/mod_curia.c	2013-02-13 00:52:07.000000000 +0900
***************
*** 233,239 ****
    const char *name;
    int index, omode, bnum, dnum;
    if((index = getnewindex()) == -1) myerror(DP_EMISC);
!   name = STR2CSTR(vname);
    FIXNUM_P(vomode);
    omode = FIX2INT(vomode);
    FIXNUM_P(vbnum);
--- 233,239 ----
    const char *name;
    int index, omode, bnum, dnum;
    if((index = getnewindex()) == -1) myerror(DP_EMISC);
!   name = StringValuePtr(vname);
    FIXNUM_P(vomode);
    omode = FIX2INT(vomode);
    FIXNUM_P(vbnum);
***************
*** 276,285 ****
    int index, ksiz, vsiz, dmode;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
!   vbuf = STR2CSTR(vval);
!   vsiz = RSTRING(vval)->len;
    FIXNUM_P(vdmode);
    dmode = FIX2INT(vdmode);
    curia = crtable[index];
--- 276,285 ----
    int index, ksiz, vsiz, dmode;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
!   vbuf = StringValuePtr(vval);
!   vsiz = RSTRING_LEN(vval);
    FIXNUM_P(vdmode);
    dmode = FIX2INT(vdmode);
    curia = crtable[index];
***************
*** 297,304 ****
    int index, ksiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    curia = crtable[index];
    if(!crout(curia, kbuf, ksiz)){
      if(crsltable[index] && dpecode == DP_ENOITEM) return Qfalse;
--- 297,304 ----
    int index, ksiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    curia = crtable[index];
    if(!crout(curia, kbuf, ksiz)){
      if(crsltable[index] && dpecode == DP_ENOITEM) return Qfalse;
***************
*** 316,323 ****
    VALUE vval;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    FIXNUM_P(vstart);
    start = FIX2INT(vstart);
    FIXNUM_P(vmax);
--- 316,323 ----
    VALUE vval;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    FIXNUM_P(vstart);
    start = FIX2INT(vstart);
    FIXNUM_P(vmax);
***************
*** 339,346 ****
    int index, ksiz, vsiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    curia = crtable[index];
    if((vsiz = crvsiz(curia, kbuf, ksiz)) == -1){
      if(crsltable[index] && dpecode == DP_ENOITEM) return INT2FIX(-1);
--- 339,346 ----
    int index, ksiz, vsiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    curia = crtable[index];
    if((vsiz = crvsiz(curia, kbuf, ksiz)) == -1){
      if(crsltable[index] && dpecode == DP_ENOITEM) return INT2FIX(-1);
