*** orig/ruby/depot/mod_depot.c	2006-09-25 21:21:38.000000000 +0900
--- patched/ruby/depot/mod_depot.c	2013-02-13 00:51:27.000000000 +0900
***************
*** 232,238 ****
    const char *name;
    int index, omode, bnum;
    if((index = getnewindex()) == -1) myerror(DP_EMISC);
!   name = STR2CSTR(vname);
    FIXNUM_P(vomode);
    omode = FIX2INT(vomode);
    FIXNUM_P(vbnum);
--- 232,238 ----
    const char *name;
    int index, omode, bnum;
    if((index = getnewindex()) == -1) myerror(DP_EMISC);
!   name = StringValuePtr(vname);
    FIXNUM_P(vomode);
    omode = FIX2INT(vomode);
    FIXNUM_P(vbnum);
***************
*** 273,282 ****
    int index, ksiz, vsiz, dmode;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
!   vbuf = STR2CSTR(vval);
!   vsiz = RSTRING(vval)->len;
    FIXNUM_P(vdmode);
    dmode = FIX2INT(vdmode);
    depot = dptable[index];
--- 273,282 ----
    int index, ksiz, vsiz, dmode;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
!   vbuf = StringValuePtr(vval);
!   vsiz = RSTRING_LEN(vval);
    FIXNUM_P(vdmode);
    dmode = FIX2INT(vdmode);
    depot = dptable[index];
***************
*** 294,301 ****
    int index, ksiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    depot = dptable[index];
    if(!dpout(depot, kbuf, ksiz)){
      if(dpsltable[index] && dpecode == DP_ENOITEM) return Qfalse;
--- 294,301 ----
    int index, ksiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    depot = dptable[index];
    if(!dpout(depot, kbuf, ksiz)){
      if(dpsltable[index] && dpecode == DP_ENOITEM) return Qfalse;
***************
*** 313,320 ****
    VALUE vval;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    FIXNUM_P(vstart);
    start = FIX2INT(vstart);
    FIXNUM_P(vmax);
--- 313,320 ----
    VALUE vval;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    FIXNUM_P(vstart);
    start = FIX2INT(vstart);
    FIXNUM_P(vmax);
***************
*** 336,343 ****
    int index, ksiz, vsiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    depot = dptable[index];
    if((vsiz = dpvsiz(depot, kbuf, ksiz)) == -1){
      if(dpsltable[index] && dpecode == DP_ENOITEM) return INT2FIX(-1);
--- 336,343 ----
    int index, ksiz, vsiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    depot = dptable[index];
    if((vsiz = dpvsiz(depot, kbuf, ksiz)) == -1){
      if(dpsltable[index] && dpecode == DP_ENOITEM) return INT2FIX(-1);
