*** orig/ruby/villa/mod_villa.c	2006-09-25 23:47:04.000000000 +0900
--- patched/ruby/villa/mod_villa.c	2013-02-13 00:52:56.000000000 +0900
***************
*** 304,310 ****
    int index, omode, cmode;
    VLCFUNC cmp;
    if((index = getnewindex()) == -1) myerror(DP_EMISC);
!   name = STR2CSTR(vname);
    FIXNUM_P(vomode);
    omode = FIX2INT(vomode);
    FIXNUM_P(vcmode);
--- 304,310 ----
    int index, omode, cmode;
    VLCFUNC cmp;
    if((index = getnewindex()) == -1) myerror(DP_EMISC);
!   name = StringValuePtr(vname);
    FIXNUM_P(vomode);
    omode = FIX2INT(vomode);
    FIXNUM_P(vcmode);
***************
*** 352,361 ****
    int index, ksiz, vsiz, dmode;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
!   vbuf = STR2CSTR(vval);
!   vsiz = RSTRING(vval)->len;
    FIXNUM_P(vdmode);
    dmode = FIX2INT(vdmode);
    villa = vltable[index];
--- 352,361 ----
    int index, ksiz, vsiz, dmode;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
!   vbuf = StringValuePtr(vval);
!   vsiz = RSTRING_LEN(vval);
    FIXNUM_P(vdmode);
    dmode = FIX2INT(vdmode);
    villa = vltable[index];
***************
*** 373,380 ****
    int index, ksiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    villa = vltable[index];
    if(!vlout(villa, kbuf, ksiz)){
      if(vlsltable[index] && dpecode == DP_ENOITEM) return Qfalse;
--- 373,380 ----
    int index, ksiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    villa = vltable[index];
    if(!vlout(villa, kbuf, ksiz)){
      if(vlsltable[index] && dpecode == DP_ENOITEM) return Qfalse;
***************
*** 391,398 ****
    VALUE vval;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    villa = vltable[index];
    if(!(vbuf = vlget(villa, kbuf, ksiz, &vsiz))){
      if(vlsltable[index] && dpecode == DP_ENOITEM) return Qnil;
--- 391,398 ----
    VALUE vval;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    villa = vltable[index];
    if(!(vbuf = vlget(villa, kbuf, ksiz, &vsiz))){
      if(vlsltable[index] && dpecode == DP_ENOITEM) return Qnil;
***************
*** 409,416 ****
    int index, ksiz, vsiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    villa = vltable[index];
    if((vsiz = vlvsiz(villa, kbuf, ksiz)) == -1){
      if(vlsltable[index] && dpecode == DP_ENOITEM) return INT2FIX(-1);
--- 409,416 ----
    int index, ksiz, vsiz;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    villa = vltable[index];
    if((vsiz = vlvsiz(villa, kbuf, ksiz)) == -1){
      if(vlsltable[index] && dpecode == DP_ENOITEM) return INT2FIX(-1);
***************
*** 426,433 ****
    int index, ksiz, vnum;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    villa = vltable[index];
    vnum = vlvnum(villa, kbuf, ksiz);
    return INT2FIX(vnum);
--- 426,433 ----
    int index, ksiz, vnum;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    villa = vltable[index];
    vnum = vlvnum(villa, kbuf, ksiz);
    return INT2FIX(vnum);
***************
*** 497,504 ****
    VALUE vval;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = STR2CSTR(vkey);
!   ksiz = RSTRING(vkey)->len;
    FIXNUM_P(vjmode);
    jmode = FIX2INT(vjmode);
    villa = vltable[index];
--- 497,504 ----
    VALUE vval;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   kbuf = StringValuePtr(vkey);
!   ksiz = RSTRING_LEN(vkey);
    FIXNUM_P(vjmode);
    jmode = FIX2INT(vjmode);
    villa = vltable[index];
***************
*** 551,558 ****
    int index, vsiz, cpmode;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   vbuf = STR2CSTR(vval);
!   vsiz = RSTRING(vval)->len;
    FIXNUM_P(vcpmode);
    cpmode = FIX2INT(vcpmode);
    villa = vltable[index];
--- 551,558 ----
    int index, vsiz, cpmode;
    FIXNUM_P(vindex);
    if((index = FIX2INT(vindex)) == -1) myerror(DP_EMISC);
!   vbuf = StringValuePtr(vval);
!   vsiz = RSTRING_LEN(vval);
    FIXNUM_P(vcpmode);
    cpmode = FIX2INT(vcpmode);
    villa = vltable[index];
